client/messaging/external-link.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import { openDialog } from '../dialogs/action-creators'
+import { DialogType } from '../dialogs/dialog-type'
+import { useAppDispatch, useAppSelector } from '../redux-hooks'
+import { RootState } from '../root-reducer'
+import { LocalSettings } from '../settings/settings-records'
+
+interface ExternalLinkProps {
+  href: string
+  innerText: string
+}
+
+const localSettingsSelector = (s: RootState) => s.settings.local
+
+export default function ExternalLink({ href, innerText }: ExternalLinkProps) {
+  const dispatch = useAppDispatch()
+  const localSettings: LocalSettings = useAppSelector(localSettingsSelector)
+
+  return (
+    <a
+      href={href}
+      target='_blank'
+      rel='noopener nofollow'
+      onClick={e => {
```

Comment:
I would do this whole thing a bit differently. Instead of having an inlined event handler here that decides whether to open a dialog or not, I would dispatch an action creator that takes the the click event as an argument, e.g.
```ts
onClick={e => dispatch(maybeOpenExternalLinkDialog(e))}
```
And then that action creator would dispatch a thunk that checks trusted hosts and decides whether to open a dialog or not, e.g.
```ts
function maybeOpenExternalLinkDialog(clickEvent: React.MouseEvent): ThunkAction {
  return (dispatch, getState) => {
    const { settings: { local: { trustedHosts } } } = getState()
    
    // ...rest of code
  }
}
```
You can create this action creator in `action-creators.ts` file in `messaging` folder. This would make it unnecessary to have the selector for local settings in this component at all.

Reply:
Is it necessary then to create `client/messaging/actions.ts` and import its types in `client/action-types.ts` ?

Reply:
Yeah, that is how we usually organize stuff now. Maybe it doesn't make much sense to create a separate file for just one action, but it's very likely we'll add more actions related to messaging in the future, so I'd just create the file now.